pandas/_libs/index.pyx

```diff
@@ -275,6 +275,12 @@ cdef class IndexEngine:
 
         self._ensure_mapping_populated()
         values = np.array(self._get_index_values(), copy=False)
+        targets_mask = isnaobj(targets)
+        values_mask = isnaobj(values)
+        if targets_mask.any():
+            targets[targets_mask] = 0
```

Comment:
we are mutating the inputs, *might* be ok, but can copy before you do this is better

Reply:
gotcha, I made a copy of values as well (since it is not a copy and we are modifying it)